integration-test/src/main/java/com/sequenceiq/it/cloudbreak/action/v1/distrox/FetchPreferredSubnetForInstanceNetworkIfMultiAzEnabledAction.java

```diff
@@ -0,0 +1,31 @@
+package com.sequenceiq.it.cloudbreak.action.v1.distrox;
+
+import java.util.ArrayList;
+
+import com.sequenceiq.distrox.api.v1.distrox.model.network.InstanceGroupNetworkV1Request;
+import com.sequenceiq.distrox.api.v1.distrox.model.network.aws.InstanceGroupAwsNetworkV1Parameters;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
+import com.sequenceiq.it.cloudbreak.CloudbreakClient;
+import com.sequenceiq.it.cloudbreak.EnvironmentClient;
+import com.sequenceiq.it.cloudbreak.action.Action;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.distrox.DistroXTestDto;
+
+public class FetchPreferredSubnetForInstanceNetworkIfMultiAzEnabledAction implements Action<DistroXTestDto, CloudbreakClient> {
```

Comment:
this seems as a util for the test, not an actual test action, right?
shouldn't be this a testDto 'withEnironmentSubnets' method rather?

Reply:
I think it is a mixed one. We fetch from the server via API and set the value. I prefer when we set a direct value via the testDto like your suggestion. We should just set the value because if we set the value (no network, no valid data) we don't need to break the run. Probably, it is not causing any problem here, but if someone uses the `.valid()` we can't override the value later. We should only break the test run via Acton or Assertion.